pkg/storegateway/bucket_index_postings.go

```diff
@@ -463,3 +465,52 @@ func (s speculativeFetchedDataStrategy) selectPostings(groups []postingGroup) (s
 	}
 	return groups, nil
 }
+
+// labelValuesPostingsStrategy works in a similar way to worstCaseFetchedDataStrategy.
+// The differences are:
+//   - it doesn't a factor for the posting list size
+//   - as the bounded maximum for fetched data it also takes into account the provided postingLists; this is useful in
+//     LabelValues calls where we have to decide between fetching (some expanded postings + series),
+//     (expanded postings + series), or (expanded postings + postings for each label value).
+type labelValuesPostingsStrategy struct {
+	delegate     postingsSelectionStrategy
+	postingLists []streamindex.PostingListOffset
+}
+
+func (w labelValuesPostingsStrategy) name() string {
+	return "lv-" + w.delegate.name()
+}
+
+func (w labelValuesPostingsStrategy) selectPostings(groups []postingGroup) (selected, omitted []postingGroup) {
+	selected, omitted = w.delegate.selectPostings(groups)
+
+	maxPossibleSeriesSize := numSeriesInSmallestIntersectingPostingGroup(selected) * tsdb.EstimatedSeriesP99Size
+
+	completePostings := postingGroupsTotalSize(groups)
+	completePostingsPlusPostings := completePostings + postingsListsTotalSize(w.postingLists)
+	completePostingsPlusSeries := completePostings + maxPossibleSeriesSize
+	partialPostingsPlusSeries := postingGroupsTotalSize(selected) + maxPossibleSeriesSize
```

Comment:
[nit] Here it's called "partial" while above it's called "selected". I would standardise on naming.